cmd/kritis/admission/main.go

```diff
@@ -87,18 +92,25 @@ func main() {
 		kritisConf := kritisConfigs[0]
 		// TODO(https://github.com/grafeas/kritis/issues/304): Use CRD validation instead
 		if kritisConf.Spec.MetadataBackend != "" {
-			metadataBackend = kritisConf.Spec.MetadataBackend
+			config.Metadata = kritisConf.Spec.MetadataBackend
 		}
 		if kritisConf.Spec.CronInterval != "" {
 			cronInterval = kritisConf.Spec.CronInterval
 		}
 		if kritisConf.Spec.ServerAddr == "" {
 			serverAddr = kritisConf.Spec.ServerAddr
 		}
-	}
-
-	config := &admission.Config{
-		Metadata: metadataBackend,
+		if config.Metadata == constants.GrafeasMetadata {
```

Comment:
Could we add an integration test to verify the new code paths?

Reply:
As far as i see, there is no coverage for this flow (extracting configuration metadata), so refactored the code to use a common validation method with appropriate unit tests in the grafeas package.